woocommerce/woocommerce
# Extract attribute filtering and fetching logic out of attribute components

### All Submissions:






### Changes proposed in this Pull Request:

Extracts all fetching and filtering of attributes to a separate hook so that the attribute component knows less about how to retrieve or filter types of attributes.

Two more follow-up PRs will be created to further remove variation logic in the attribute components and optimize fetching terms.

Closes #36304  .






### How to test the changes in this Pull Request:



1. Navigate to the new products experience (Products-> Add new (MVP))
2. Add some attributes and/or options
3. Edit some attributes
4. Remove some attributes
5. Toggle between tabs to make sure data between options/attributes does not affect the other
6. Make sure no regressions occur



### Other information:




### FOR PR REVIEWER ONLY:
